addons/mail/static/src/models/thread_viewer/thread_viewer.js

```diff
@@ -97,15 +83,24 @@ function factory(dependencies) {
 
         /**
          * @private
-         * @returns {mail.thread_cache|undefined}
+         * @returns {mail.thread_cache}
          */
         _computeThreadCache() {
-            if (!this.thread) {
-                return [];
-            }
+            this.addComponentHint('change-of-thread-cache');
```

Comment:
No guarantee that compute method is called once during an update cycle.
So this will insert lots of such hints.
Note that the management of component hints is too naïve right now, like marking a hint as processed removes all occurrences matching the name of hint...

Reply:
Isn't that exactly what we want? Feels like hint should be made a `set` instead of a `list` then?